autogen/main/sa.tf.tmpl

```diff
@@ -39,7 +42,7 @@ resource "random_string" "cluster_service_account_suffix" {
 resource "google_service_account" "cluster_service_account" {
   count        = var.create_service_account ? 1 : 0
   project      = var.project_id
-  account_id   = "tf-gke-${substr(var.name, 0, min(15, length(var.name)))}-${random_string.cluster_service_account_suffix.result}"
+  account_id   = local.service_account_name
```

Comment:
Do we need `local.service_account_name`, or would this be simpler?
```suggestion
  account_id   = var.service_account_name == "" ? local.service_account_default_name : var.service_account_name
```

Reply:
Indeed I'm removing it ! 

I wanted to use var.service_account at the beginning but this variable contains the full email of the service account, whereas here we just want the name. Maybe we should make the doc clearer here ?

Reply:
Sure, a code comment or something else might be nice.  I'll re-trigger the CI.

/gcbrun

Reply:
I've just pushed an update of the doc, you might need to retrigger the CI, sorry